jwst/pipeline/calwebb_guider.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+from ..stpipe import Pipeline
+from .. import datamodels
+import os
+
+# step imports
+from ..dq_init import dq_init_step
+from ..flatfield import flat_field_step
+from ..guider_cds import guider_cds_step
+
+__version__ = "0.7.1"
+
+# Define logging
+import logging
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+class GuiderPipeline(Pipeline):
+    """
+    GuiderPipeline: For FGS observations, apply all calibration 
+    steps to raw JWST ramps to produce a 3-D slope product. 
+    Included steps are: dq_init, flat_field, and guider_cds.
+    """
+
+    # Define aliases to steps
+    step_defs = {'dq_init': dq_init_step.DQInitStep,
+                 'flat_field': flat_field_step.FlatFieldStep,
+                 'guider_cds': guider_cds_step.GuiderCdsStep,
+                 }
+
+    # start the actual processing
+    def process(self, input):
+
+        log.info('Starting calwebb_guider ...')
+
+        # open the input
+        input = datamodels.GuiderRawModel(input)   # change  to 'input/Raw' type-  071817 02:50
+
+        input = self.dq_init(input)  
+        input = self.flat_field(input)
```

Comment:
It would be easiest to call flat_field after the guider_cds step, just as we do in normal calwebb_sloper, calimage2 processing flow, so that the flat_field step only has to work on the countrate images, rather than being applied to each of the raw groups. Speaking of which, it'll be interesting to see if flat_field is able to handle the new data models as input, or whether it will also need updating, like dq_init.